src/test/run-make/export-executable-symbols/main.rs

```diff
@@ -0,0 +1,38 @@
+// edition:2018
+
+#![feature(rustc_private)]
+
+extern crate libc;
+use std::ffi::*;
+use std::os::unix::ffi::*;
+
+fn main() {
+    let path = std::env::var("TMPDIR").unwrap();
+    let path = std::path::PathBuf::from(path).join("libfoo.so");
+
+    let s = CString::new(path.as_os_str().as_bytes()).unwrap();
+    let handle = unsafe { libc::dlopen(s.as_ptr(), libc::RTLD_LAZY | libc::RTLD_LOCAL) };
```

Comment:
I think this test will need to be restricted to unix targets or unix and windows targets if the windows api is used. Wasm can't support this either way.

Reply:
Sounds reasonable but I'm not sure how to do that!

Reply:
Probably some annotation on the test file?

Reply:
The `Makefile` accepts the same test flags as rust files in `test/ui`.